session/session.go

```diff
@@ -144,6 +145,7 @@ type Session interface {
 	Auth(user *auth.UserIdentity, auth []byte, salt []byte) bool
 	AuthWithoutVerification(user *auth.UserIdentity) bool
 	ShowProcess() *util.ProcessInfo
+	TxnInfo() []types.Datum
```

Comment:
Would it be better if we abstract a type `TxnInfo` here?

Pros:

- The order of fields are no longer important, which makes testing easier

Cons:

- Adds one more struct `TxnInfo`, which is somehow easy to confuse with `TxnState`

Maybe  `TxnState` can have a better name like `LazyTxn`?